### PR TITLE
Refine Keras-like API LeNet definition

### LINKED ISSUE #2402: Refactor model definition under example/keras/LeNet
The current LeNet definition is not correct. we can take it as a cnn model, but not the same with the original LeNet definition. Refer to model/lenet.

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

1. Correct original Torch definition of LeNet.
2. Sync lenet definition with Keras-Style API with original Torch-Style API.
3. Update the docs and readme accordingly.
4. Also fix a conflict of set_creator_class
5. KerasLayerWrapper compute output shape use cloneModule() and input use tensor.fill(ev.one).

## How was this patch tested?

Manual test by running lenet.

## Related links or issues (optional)
fixed #2402 #2411 #2305

### CODE REVIEW

File: docs/docs/KerasStyleAPIGuide/keras-api-python.md [hunk 1]
```diff
@@ -119,19 +119,19 @@ from bigdl.nn.keras.layer import *
...
-model.add(Flatten())
-model.add(Dense(128, activation="relu"))
-model.add(Dropout(0.5))
-model.add(Dense(10, activation="softmax"))
+model.add(Convolution2D(6, 5, 5, activation="tanh", name="conv1_5x5"))
+model.add(MaxPooling2D())
+model.add(Activation("tanh"))
+model.add(Convolution2D(12, 5, 5, name="conv2_5x5"))
+model.add(MaxPooling2D())
+model.add(Reshape((12 * 4 * 4, )))
```
Comment (zhichao-li) [inline docs/docs/KerasStyleAPIGuide/keras-api-python.md]: replace `Reshape` with `Flatten()`

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/401/

**Comment by zhichao-li** [comment]
LGTM

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/models/lenet/LeNet5.scala [hunk 2]
```diff
@@ -27,23 +27,24 @@ object LeNet5 {
       .add(SpatialConvolution(1, 6, 5, 5).setName("conv1_5x5"))
       .add(Tanh())
       .add(SpatialMaxPooling(2, 2, 2, 2))
-      .add(Tanh())
       .add(SpatialConvolution(6, 12, 5, 5).setName("conv2_5x5"))
+      .add(Tanh())
```
Comment (hkvision) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/models/lenet/LeNet5.scala]: Changing the model definition will cause some unit tests fail.

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/417/

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/job/BigDL-PR-Validation-v3/418/

**Pull request merged**